core/dbt/contracts/graph/metrics.py

```diff
@@ -17,7 +24,7 @@ class ResolvedMetricReference(MetricReference):
     for working with metrics (ie. __str__ and templating functions)
     """
 
-    def __init__(self, node, manifest, Relation):
+    def __init__(self, node: Metric, manifest: Manifest, Relation=None):
```

Comment:
nit: `Relation` is not typed. we could also include the `-> None` return type here to avoid mypy warnings such as the ones cleaned up in https://github.com/dbt-labs/dbt-core/issues/8387.

Related, but not introduced in this PR, so could be addressed in a follow-on: it feels pretty nonstandard to have this variable name be capitalized - it looks like this is propagated from `db_wrapper.Relation` in `RuntimeMetricResolver` -- perhaps we could rename to `relation_cls` instead?

Reply:
Happy to type `Relation`  and rename it , as well as add the `-> None:`. Interestingly `Relation` isn't used at all in this class and thus far it seems dbt-core instantiations of this class pass `None` as the third variable. I wanted to rip it out very badly while looking at this code, but didn't want to break any third parties instantiating the class for whatever reason.